src/data/DataSetExecutionJsonRepository.ts

```diff
@@ -0,0 +1,18 @@
+import fs from "fs";
+import { Async } from "domain/entities/Async";
+import { DataSetExecutionRepository } from "domain/repositories/DataSetExecutionRepository";
+import { NotificationStore } from "domain/usecases/SendNotificationDataValuesUseCase";
+import { SettingsOptions } from "domain/repositories/SettingsRepository";
+
+export class DataSetExecutionJsonRepository implements DataSetExecutionRepository {
+    async save(value: NotificationStore, options: SettingsOptions): Async<void> {
+        fs.writeFileSync(options.path, JSON.stringify(value, null, 4));
+    }
+
+    async get(options: SettingsOptions): Async<NotificationStore | undefined> {
+        const fileContent = fs.readFileSync(options.path, "utf-8");
+        if (!fileContent) return undefined;
```

Comment:
The executions file won't exist on first run and that should not be an error (now readFileSync fails)